addons/stock/models/stock_inventory.py

```diff
@@ -213,7 +213,7 @@ def action_inventory_line_tree(self):
     def _get_inventory_lines_values(self):
         # TDE CLEANME: is sql really necessary ? I don't think so
         locations = self.env['stock.location'].search([('id', 'child_of', [self.location_id.id])])
-        domain = ' location_id in %s'
+        domain = 'product_product.id = stock_quant.product_id AND location_id in %s AND product_product.active = TRUE'
```

Comment:
Shouldn't you do a proper `JOIN` rather than this?

Reply:
WHERE is similar than an INNER JOIN but 'active' use a default value and should always be set and 'product_id' is required so it should be enough. However I could modify it to explicit JOIN or LEFT JOIN to be sure.